src/clarity-angular/datagrid/_datagrid.clarity.scss

```diff
@@ -107,34 +106,33 @@
 
     $clr-datagrid-icon-size: rem(1);
 
-    .datagrid {
-        flex: 1 1 auto;
-        width: auto; // override the basic table's style of width: 100%
-        max-width: none; // override the basic table's style of max-width: 100%
+    .datagrid-overlay-wrapper {
+        display: flex;
+        flex-direction: row;
+        flex: 0 0 100%;
+        border: 2px solid darkgreen;
+        height: 100%;
 
         .datagrid-spinner-wrapper {
-            position: relative;
+            border: 2px solid darkblue;
+            flex: 0 0 100%;
+            margin-left: -100%;
+            display: flex;
+            justify-content: center;
+            align-items: center;
 
             .datagrid-spinner {
-                position: absolute;
-                top: 0;
-                bottom: 0;
-                left: 0;
-                right: 0;
                 background: rgba($clr-white, 0.6);
-
-                .spinner {
-                    // We center the spinner in the datagrid
-                    position: absolute;
-                    top: 0;
-                    bottom: 0;
-                    left: 0;
-                    right: 0;
-                    margin: auto;
-                }
             }
         }
 
+    }
+
+    .datagrid {
+        flex: 1 1 auto;
+        width: auto; // override the basic table's style of width: 100%
+        max-width: none; // override the basic table's style of max-width: 100%
```

Comment:
Same thing here, that was an existing mistake be the layout styles of `.datagrid` should go line 21.